test/tests/feedItemTest.js

```diff
@@ -92,18 +92,35 @@ describe("Zotero.FeedItem", function () {
 	});
 	describe("#fromJSON()", function() {
 		it("should attempt to parse non ISO-8601 dates", function* () {
-			var json = {
-				itemType: "journalArticle",
-				accessDate: "2015-06-07 20:56:00",
-				dateAdded: "18-20 June 2015", // magically parsed by `new Date()`
-				dateModified: "07/06/2015", // US
-			};
-			var item = new Zotero.FeedItem;
-			item.fromJSON(json);
-			assert.strictEqual(item.getField('accessDate'), '2015-06-07 20:56:00');
-			assert.strictEqual(item.getField('dateAdded'), '2015-06-18 20:00:00');
-			// sets a timezone specific hour when new Date parses from strings without hour specified.
-			assert.strictEqual(item.getField('dateModified'), Zotero.Date.dateToSQL(new Date(2015, 6, 6), true)); 
+			var data = [
+				{
+					itemType: "journalArticle",
+					date: "2015-06-07 20:56:00" // sql
+				},
+				{
+					itemType: "journalArticle",
+					date: "Mon, 13 Jun 2016 06:25:57 EDT" // HTTP
+				},
+				{
+					itemType: "journalArticle",
+					date: "18-20 June 2015" // magically parsed by `new Date()`
+				},
+				{
+					itemType: "journalArticle",
+					date: "07/06/2015" // american format
+				}
+			];
+			var expectedDates = [
+				'2015-06-07 20:56:00',
+				'2016-06-13 10:25:57',
+				'2015-06-18 20:00:00',
```

Comment:
This is wrong, though — the 20 is being picked up as a time. This shouldn't actually be parsed by new Date — we should add a Zotero.Date.isHTTP() and limit the parsing to that. Otherwise it should use strToDate(), which (I think) should get June 18, 2015.